app/uk/gov/hmrc/economiccrimelevyaccount/services/FinancialDataService.scala

```diff
@@ -53,7 +53,10 @@ class FinancialDataService @Inject() (
               outstandingAmount,
               extractValue(firstLineItemDetailsElement.periodFromDate),
               extractValue(firstLineItemDetailsElement.periodToDate),
-              extractValue(firstLineItemDetailsElement.periodKey),
+              value.getPaymentType match {
```

Comment:
In scala it's preferred to return a None rather than an empty string, is it too late to make the parameter optional?

Reply:
I suggest we tackle this as a part of the tech debt process :) .